ansible_collections/arista/avd/roles/eos_designs/python_modules/l3_edge/utils.py

```diff
@@ -0,0 +1,376 @@
+            return p2p_link
+
+        if "subnet" in p2p_link:
+            # Resolve ip from subnet
+            subnet = ip_network(p2p_link["subnet"], strict=False)
+
+        elif "ip_pool" not in p2p_link or "id" not in p2p_link or not self._p2p_links_ip_pools:
+            # Subnet not set and not possible to resolve from pool. Returning original
+            return p2p_link
+
+        else:
+            # Resolving subnet from pool
+            ip_pool = get_item(self._p2p_links_ip_pools, "name", p2p_link["ip_pool"], default={}).get("ipv4_pool")
+            if not ip_pool:
+                # Not possible to resolve from pool. Returning original
+                return p2p_link
+            id = int(p2p_link["id"])
+            subnet = list(islice(ip_network(ip_pool).subnets(new_prefix=31), id - 1, id))[0]
+
+        # hosts() return an iterator of all hosts in subnet.
+        # islice() return a generator with only the first two iterations of hosts.
+        # List comprehension runs through the generator creating string from each.
+        p2p_link["ip"] = [f"{ip}/{subnet.prefixlen}" for ip in islice(subnet.hosts(), 2)]
+        return p2p_link
+
+    def _get_p2p_data(self, p2p_link: dict) -> dict:
+        """
+        Parses p2p_link data model and extracts information which is easier to parse.
+        Returns:
+        {
+            peer: <peer name>
+            peer_type: <type of peer>
+            interface: <interface on this node>
+            peer_interface: <interface on peer>
+            port_channel_id: <id on this node | None>
+            port_channel_members:
+              - interface: <interface on this node>
+                peer_interface: <interface on peer>
+            ip: <ip if set | None>
+            peer_ip: <peer ip if set | None>
+            bgp_as: <as if set | None>
+            peer_bgp_as: <peer as if set | None>
+        }
+        """
+        index = p2p_link["nodes"].index(self._hostname)
+        peer_index = (index + 1) % 2
+        peer = p2p_link["nodes"][peer_index]
+        peer_type = get(self._hostvars, f"avd_switch_facts..{peer}..switch..type", default="other", separator="..")
+
+        # Set ip or fallback to list with None values
+        ip = get(p2p_link, "ip", default=[None, None])
+        # Set bgp_as or fallback to list with None values
+        bgp_as = get(p2p_link, "as", default=[None, None])
+
+        data = {
+            "peer": peer,
+            "peer_type": peer_type,
+            "ip": ip[index],
+            "peer_ip": ip[peer_index],
+            "bgp_as": str(bgp_as[index]) if bgp_as[index] is not None else None,
+            "peer_bgp_as": str(bgp_as[peer_index]) if bgp_as[peer_index] is not None else None,
+        }
+
+        node_child_interfaces = get(p2p_link, "port_channel.nodes_child_interfaces")
+        # Convert to new data models
+        node_child_interfaces = convert_dicts(node_child_interfaces, primary_key="node", secondary_key="interfaces")
```

Comment:
```suggestion
        # Convert to new data models
        node_child_interfaces = convert_dicts(get(p2p_link, "port_channel.nodes_child_interfaces"), primary_key="node", secondary_key="interfaces")
```

Reply:
I prefer to keep it in a separate line, because it will be removed when we get the schema conversion in eos_designs

Reply:
okay